apps/src/templates/manageStudents/Table/UsStateColumn/BulkSetModal/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React, {useState} from 'react';
+import {ControlLabel, FormControl, FormGroup} from 'react-bootstrap';
+import {connect} from 'react-redux';
+
+import {Heading4} from '@cdo/apps/componentLibrary/typography';
+import {STATE_CODES} from '@cdo/apps/geographyConstants';
+import StylizedBaseDialog from '@cdo/apps/sharedComponents/StylizedBaseDialog';
+import {bulkSet} from '@cdo/apps/templates/manageStudents/manageStudentsRedux';
+import SafeMarkdown from '@cdo/apps/templates/SafeMarkdown';
+import {CapLinks} from '@cdo/generated-scripts/sharedConstants';
+import i18n from '@cdo/locale';
+
+import './style.scss';
+
+interface BulkSetModalProps {
+  isOpen?: boolean;
+  initVal?: string;
+  onClose: () => void;
+  bulkSet: (studentsData: {usState: string | null}) => void;
+}
+
+const BulkSetModal: React.FC<BulkSetModalProps> = ({
+  isOpen = false,
+  initVal = '',
+  onClose = () => {},
+  // Provided by redux
+  bulkSet,
+}) => {
+  const [usState, setUsState] = useState(initVal);
+
+  const handleUsStateChange: React.FormEventHandler<FormControl> = event => {
+    setUsState((event.target as HTMLInputElement).value);
+  };
+
+  const bulkSetStudents = () => {
+    bulkSet({usState: usState || null});
+    onClose();
```

Comment:
If `onClose` is optional, we can do 

```suggestion
    if (typeof onClose === 'function')
      onClose();
    }
```